sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c

```diff
@@ -33,6 +33,11 @@
 #include "debug.h"
 
 /**
+ * @brief Flag indicating if a thread blocked by the GNRC TCP Api must be notified.
+ */
+static bool _notify_user;
```

Comment:
But then you need to differ between additionally between the triggered event and the returned values.
In cases there the Send or Receive Event was triggered the return value would mean the number of sent/received bytes in any other cases it would mean that the caller must be notified. 

I think in that case a global variable (only accessible in the translation unit) is the clearest way to do that. Concurrency isn't an issue either because the whole fsm is locked via a mutex.